app/src/main/java/org/squidfish/tuplenback/games/engines/GameEngine.kt

```diff
@@ -37,13 +37,15 @@ abstract class GameEngine(recallsBack: Int, private val repeatChance: Int) {
      *
      * @see[createNewState]
      */
-    protected abstract fun genNewMnemonic(forbidden: List<Int?> = emptyList()): Int
+    protected abstract fun genNewMnemonic(forbidden: List<Int> = emptyList()): Int
+
+    abstract fun onNewRound(mnemonicId: Int)
 
     /**
      * Determine if the next generated mnemonic should be a repeat
      * @return [repeatChance]% chance of returning True
      */
-    private fun shouldRepeat(): Boolean = Random.Default.nextInt(1, 100) <= repeatChance
+    private fun shouldRepeat(): Boolean = Random.Default.nextInt(100) <= repeatChance
```

Comment:
```suggestion
    private fun shouldRepeat(): Boolean = Random.Default.nextInt(100) < repeatChance
```
You might have forgotten.. as discussed, this has the problem of a slightly biased (towards repetition) distribution. The numbers that will be compared with `repeatChance` are within the range [0, 100). Assume that repeat chance is 0, although this is true for any number. Then, if the random integer is also 0, the check
`Random.Default.nextInt(100) <= repeatChance` is equivalent to `0 <= 0`, which is true, thereby having a 1% repeat chance when 0 is expected.

Reply:
Yup, that was commited by accident because I never reverted the change 😅